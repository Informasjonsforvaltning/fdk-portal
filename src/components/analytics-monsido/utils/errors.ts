export const MonsidoError = {
  UnknownAction: 'Monsido: unknown action',
  MonsidoFunctionsNotDefined: 'Monsido: monsido_functions is not defined',
  EnvironmentVariableNotDefined: 'Monsido: environment variable is not defined'
};
