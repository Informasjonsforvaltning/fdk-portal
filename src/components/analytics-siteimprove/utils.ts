export enum EventCategory {
  DETAILS_PAGE = 'Details page'
}

export enum EventAction {
  REQUEST_ACCESS = 'Request access'
}

type SiteImproveEventProps = {
  category: EventCategory;
  action: EventAction;
  label?: string | undefined;
};

export const trackSiteImproveEvent = ({
  category,
  action,
  label
}: SiteImproveEventProps) => {
  if (window._sz === undefined) {
    // eslint-disable-next-line no-console
    console.error('Unable to find Site Improve event library.');
  } else if (label) {
    window._sz.push(['event', category, action, label]);
  } else {
    window._sz.push(['event', category, action]);
  }
};
