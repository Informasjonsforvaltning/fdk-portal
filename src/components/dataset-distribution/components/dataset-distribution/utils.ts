import { DataFormat } from '../../../../types/enums';

const priorities: { [key: string]: number } = {
  [DataFormat.JSON]: 1,
  [DataFormat.XML]: 2,
  [DataFormat.CSV]: 3
};

export const formatSorter = (a: DataFormat, b: DataFormat): number => {
  if (!(a in priorities)) {
    return 1;
  }
  if (!(b in priorities)) {
    return -1;
  }
  return priorities[a] > priorities[b] ? 1 : -1;
};
