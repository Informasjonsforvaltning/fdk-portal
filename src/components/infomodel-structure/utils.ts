import { InformationModelProperty, TextLanguage } from '../../types';

const applyRelationalTitle = (
  hasObjectType?: string | null,
  title?: Partial<TextLanguage> | null
) => {
  const parentObject = hasObjectType?.split('#').pop();
  const objectString = parentObject
    ? `${parentObject.charAt(0).toUpperCase() + parentObject.slice(1)}: `
    : '';
  return title
    ? Object.entries(title).reduce(
        (previous, [key, value]) => ({
          ...previous,
          [key]: objectString + value
        }),
        {} as Partial<TextLanguage>
      )
    : title;
};

export const resolvePropertySymmetries = (
  modelProperties: Record<string, Partial<InformationModelProperty>>
) =>
  Object.entries(modelProperties).reduce((previous, [key, value]) => {
    if (value.formsSymmetryWith) {
      return {
        ...previous,
        [key]: {
          ...value,
          title: applyRelationalTitle(
            modelProperties[value.formsSymmetryWith].hasObjectType,
            value.title
          )
        },
        [value.formsSymmetryWith]: {
          ...modelProperties[value.formsSymmetryWith],
          formsSymmetryWith: key,
          title: applyRelationalTitle(
            value.hasObjectType,
            modelProperties[value.formsSymmetryWith].title
          )
        }
      };
    }

    return { ...previous, [key]: value };
  }, {} as Record<string, Partial<InformationModelProperty>>);
