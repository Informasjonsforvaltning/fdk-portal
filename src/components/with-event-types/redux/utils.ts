import { Namespace, Fetcher, graph, isLiteral, Literal } from 'rdflib';

import type { EventType } from '../../../types';
import { getConfig } from '../../../config';

const rdf = Namespace('http://www.w3.org/1999/02/22-rdf-syntax-ns#');
const skos = Namespace('http://www.w3.org/2004/02/skos/core#');

export const getEventTypes = async (): Promise<EventType[]> => {
  const fetcher = new Fetcher(graph(), {});

  await fetcher.load(`${getConfig().searchHost.host}/api/events`, {
    withCredentials: false
  });

  const eventTypes = fetcher.store.statementsMatching(
    null,
    rdf('type'),
    skos('Concept')
  );

  return eventTypes.map((item: any) => ({
    uri: item.subject.value ?? '',
    prefLabel: fetcher.store
      .statementsMatching(item.subject, skos('prefLabel'))
      .map(({ object }) => object as Literal)
      .filter(isLiteral)
      .reduce(
        (previous, { value, language }) => ({
          ...previous,
          [language]: value
        }),
        {}
      ),
    broader: fetcher.store
      .statementsMatching(item.subject, skos('broader'))
      .map(({ object }) => object.value)
  }));
};
