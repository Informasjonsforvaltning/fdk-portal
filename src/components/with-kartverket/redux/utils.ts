import {
  Namespace,
  Fetcher,
  graph,
  isLiteral,
  isNamedNode,
  Literal
} from 'rdflib';

import type { AdministrativeUnit } from '../../../types';
import { AdministrativeUnitType } from '../../../types/enums';

const owl = Namespace('http://www.w3.org/2002/07/owl#');
const rdf = Namespace('http://www.w3.org/1999/02/22-rdf-syntax-ns#');
const rdfs = Namespace('http://www.w3.org/2000/01/rdf-schema#');

export const getAdministrativeUnit = async (
  uri: string
): Promise<AdministrativeUnit> => {
  const fetcher = new Fetcher(graph(), { withCredentials: false });

  await fetcher.load(
    `https://rdf.kartverket.no/api/1.0/adminstrative_unit/describe?uri=${uri}&mode=CBD&format=text/turtle`,
    {}
  );

  const subject = fetcher.store.anyStatementMatching(
    null,
    rdf('type'),
    owl('NamedIndividual')
  )?.subject;

  return {
    uri: subject?.value ?? '',
    name: fetcher.store
      .statementsMatching(subject, rdfs('label'))
      .map(({ object }) => object as Literal)
      .filter(isLiteral)
      .reduce(
        (previous, { value, language }) => ({
          ...previous,
          [language]: value
        }),
        {}
      ),
    type: fetcher.store
      .statementsMatching(subject, rdf('type'))
      .map(({ object }) => object)
      .filter(isNamedNode)
      .filter(({ value }) =>
        Object.values(AdministrativeUnitType).includes(value as any)
      )
      .pop()?.value as AdministrativeUnitType
  };
};
