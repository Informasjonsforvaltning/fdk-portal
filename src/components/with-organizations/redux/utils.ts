import { List, Map } from 'immutable';

import { getTranslateText as translate } from '../../../lib/translateText';

import { SortOrder } from '../../../types/enums';

export const sortOrganizations =
  (selector: string[], order: SortOrder = SortOrder.ASC) =>
  (organizations: List<any>) =>
    organizations.sortBy(
      (organization: Map<string, any>) =>
        organization.getIn(selector) ?? organization.get('name'),
      (a?: any, b?: any) =>
        (isNaN(a) && isNaN(b)
          ? translate(Map.isMap(a) ? a.toJS() : a)?.localeCompare(
              translate(Map.isMap(b) ? b.toJS() : b),
              'nb'
            ) ?? 0
          : a - b) * (order === SortOrder.ASC ? 1 : -1)
    );
