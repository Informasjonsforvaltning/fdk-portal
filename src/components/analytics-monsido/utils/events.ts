/* eslint-disable eslint-comments/disable-enable-pair */
/* eslint-disable no-console */

import { MonsidoError } from './errors';
import { getEnvironment } from './helpers';

export const MonsidoTrackEventParams = {
  Action: {
    Click: {
      Button: {
        AccessRequest: `${getEnvironment()}Access request button`
      }
    }
  },
  Category: {
    ClickTracking: `${getEnvironment()}ClickTracking`
  },
  Event: {
    ButtonClicked: `${getEnvironment()}ButtonClickEvent`
  },
  IncreaseCountBy: 1
};

export const monsidoTrackButtonClickEvent = (action: string) => {
  if (window.monsido_functions === undefined) {
    console.error(MonsidoError.MonsidoFunctionsNotDefined);
    return null;
  }
  if (
    !Object.values(MonsidoTrackEventParams.Action.Click.Button).includes(action)
  ) {
    console.error(MonsidoError.UnknownAction);
    return null;
  }
  return window.monsido_functions.trackEvent(
    MonsidoTrackEventParams.Category.ClickTracking,
    action,
    MonsidoTrackEventParams.Event.ButtonClicked,
    MonsidoTrackEventParams.IncreaseCountBy
  );
};
