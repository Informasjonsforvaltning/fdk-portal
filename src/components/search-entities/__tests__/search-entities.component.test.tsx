import React from 'react';
import { cleanup } from '@testing-library/react';
import { shallow } from 'enzyme';

import { Expectation } from '../../../../test/utils';
import response from './mock-entities-response.json';
import SearchEntities from '../search-entities.component';
import { extractEntities } from '../../../api/search-fulltext-api/all-entities';

afterEach(cleanup);

describe('SearchEntities', () => {
  const entities = extractEntities(response);
  it(Expectation.STRUCTURE, () => {
    const container = shallow(<SearchEntities entities={entities} />);
    expect(container).toMatchSnapshot();
  });
});
