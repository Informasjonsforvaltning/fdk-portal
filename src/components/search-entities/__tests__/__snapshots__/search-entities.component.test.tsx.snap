// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchEntities must render correct structure 1`] = `
<Component
  fallback={[Function]}
>
  <SearchEntities
    entities={
      Array [
        Object {
          "altLabel": Array [],
          "application": Array [],
          "definition": Object {
            "source": Object {
              "prefLabel": Object {
                "nb": "Gangsperre vs stølhet",
              },
            },
            "sourceRelationship": "basertPåKilde",
            "sources": Array [
              Object {
                "text": Object {
                  "nb": "Gangsperre vs stølhet",
                },
                "uri": "",
              },
            ],
            "text": Object {
              "nb": "Vestlandet sin egen måte å si at man er støl i muskelen på.",
            },
          },
          "harvest": Object {
            "firstHarvested": "2020-03-27T06:43:11.366+0000",
            "lastHarvested": "2020-03-29T12:02:32.252+0000",
          },
          "hiddenLabel": Array [],
          "id": "6a9aa3ce-c01b-4b2f-a289-a27b887b9a1c",
          "identifier": "https://registrering-begrep-api.staging.fellesdatakatalog.digdir.no/910244132/875e167a-ff15-4d47-92f2-14e42f4a77f4",
          "prefLabel": Object {
            "nb": "Gangsperr",
          },
          "publisher": Object {
            "name": "RAMSUND OG ROGNAN REVISJON",
            "orgPath": "/ANNET/910244132",
            "uri": "http://data.brreg.no/enhetsregisteret/enhet/910244132",
          },
          "seeAlso": Array [],
          "type": "concept",
          "uri": "https://fellesdatakatalog.brreg.no/api/concepts/6a9aa3ce-c01b-4b2f-a289-a27b887b9a1c",
        },
        Object {
          "accessRights": Object {
            "code": "PUBLIC",
            "prefLabel": Object {
              "en": "Public",
              "nb": "Offentlig",
              "nn": "Offentlig",
            },
            "uri": "http://publications.europa.eu/resource/authority/access-right/PUBLIC",
          },
          "accrualPeriodicity": Object {
            "code": "WEEKLY_3",
            "prefLabel": Object {
              "bg": "три пъти на седмица",
              "cs": "třikrát týdně",
              "da": "tre gange ugentligt",
              "de": "dreimal in Woche",
              "el": "τριεβδομαδιαίος",
              "en": "three times a week",
              "es": "tres veces por semana",
              "et": "kolm korda nädalas",
              "fi": "kolme kertaa viikossa",
              "fr": "trois fois par semaine",
              "ga": "trí uair in aghaidh na seachtaine",
              "hr": "triput tjedno",
              "hu": "hetente háromszori",
              "it": "tre volte a settimana",
              "lt": "tris kartus per savaitę",
              "lv": "trīsreiz nedēļā",
              "mt": "tliet darbiet fil-ġimgħa",
              "nl": "drie keer per week",
              "no": "tre ganger i uken",
              "pl": "trzy razy w tygodniu",
              "pt": "três vezes por semana",
              "ro": "de trei ori pe săptămână",
              "sk": "trikrát týždenne",
              "sl": "trikrat tedensko",
              "sv": "tre gånger per vecka",
            },
            "uri": "http://publications.europa.eu/resource/authority/frequency/WEEKLY_3",
          },
          "catalog": Object {
            "id": "910244132",
            "publisher": Object {
              "id": "910244132",
              "name": "RAMSUND OG ROGNAN REVISJON",
              "orgPath": "/ANNET/910244132",
              "type": "no.dcat.datastore.domain.dcat.Publisher",
              "uri": "http://data.brreg.no/enhetsregisteret/enhet/910244132",
              "valid": false,
            },
            "title": Object {
              "nb": "Datakatalog for RAMSUND OG ROGNAN REVISJON",
            },
            "uri": "http://dataset-catalogue:8080/catalogs/910244132",
          },
          "conformsTo": Array [
            Object {
              "extraType": "http://purl.org/dc/terms/Standard",
              "prefLabel": Object {
                "nb": "Test for standard",
              },
              "uri": "https://test.no",
            },
          ],
          "contactPoint": Array [],
          "description": Object {
            "nb": "Test åpne data",
          },
          "descriptionFormatted": Object {
            "nb": "Test åpne data",
          },
          "distribution": Array [
            Object {
              "accessService": Object {
                "description": Object {
                  "nb": "Sindres nasjonale opplæringskontorregister API",
                },
                "endpointDescription": Array [
                  Object {
                    "extraType": "http://xmlns.com/foaf/0.1/Document",
                    "uri": "baaeeaf2-a2d0-44d0-a5a9-d040a66993e2",
                  },
                ],
                "uri": "http://dataset-catalogue:8080/catalogs/55154089-9251-4f42-a7b7-fe0f41d270c1",
              },
              "accessURL": Array [],
              "downloadURL": Array [],
              "openLicense": false,
              "type": "API",
            },
            Object {
              "accessService": Object {
                "description": Object {
                  "nb": "Accounts API SPAREBANK 1 TELEMARK",
                },
                "endpointDescription": Array [
                  Object {
                    "extraType": "http://xmlns.com/foaf/0.1/Document",
                    "uri": "282b4993-3d8e-4171-b88d-034a88118c27",
                  },
                ],
                "uri": "http://dataset-catalogue:8080/catalogs/8d60b744-6b1d-4f26-bcb1-50875b4d159d",
              },
              "accessURL": Array [],
              "downloadURL": Array [],
              "openLicense": false,
              "type": "API",
            },
            Object {
              "accessURL": Array [
                "https://data.norge.no/nlod/en/2.0",
              ],
              "description": Object {
                "nb": "TEST TEST",
              },
              "downloadURL": Array [],
              "format": Array [
                "application/xml",
                "application/json",
              ],
              "openLicense": false,
              "page": Array [
                Object {
                  "extraType": "http://xmlns.com/foaf/0.1/Document",
                  "uri": "https://www.db.no",
                },
              ],
              "title": Object {
                "nb": "Test åpne data",
              },
              "type": "API",
            },
          ],
          "expandedLosTema": Array [
            "samfunnstryggleik og beredskap",
            "samfunnssikkerhet og beredskap",
            "civil protection and emergency planning",
            "brann",
            "brann",
            "fire",
            "forsvaret",
            "forsvaret",
            "the defense",
            "varslingstenester",
            "varslingstjenester",
            "fire",
            "bekymringsmelding om branntryggleik",
            "bekymringsmelding om brannsikkerhet",
            "fire safety report",
            "branntilsyn",
            "branntilsyn",
            "fire supervision",
            "fyrverkeri",
            "fyrverkeri",
            "fireworks",
            "feiing",
            "feiing",
            "chimney sweep services",
            "melding om overnatting",
            "melding om overnatting",
            "notice of accommodation",
            "verneplikt",
            "verneplikt",
            "compulsory military service",
            "sesjon",
            "sesjon",
            "session",
            "repetisjonsteneste",
            "repetisjonstjeneste",
            "repetition service",
            "skredvarsel",
            "skredvarsel",
            "avalanche warning",
            "blåskjelvarsel",
            "blåskjellvarsel",
            "mussels warning",
            "skogbrannvarsel",
            "skogbrannvarsel",
            "forest fire warning",
            "flyalarm",
            "flyalarm",
            "alert",
            "lynvarsel",
            "lynvarsel",
            "warning of lightning",
            "flaumvarsel",
            "flomvarsel",
            "flood warning",
            "vêrvarsel",
            "værvarsel",
            "weather forecast",
            "mobilitetstilbod",
            "mobilitetstilbud",
            "mobility services",
            "trafikk og transport",
            "trafikk og transport",
            "traffic and transport",
            "drivstoff og ladestasjonar",
            "drivstoff og ladestasjoner",
            "filling and charging stations",
            "reisebillett",
            "reisebillett",
            "travel tickets",
            "lokasjonstenester",
            "lokasjonstjenester",
            "location information",
            "skuleskyss",
            "skoleskyss",
            "school transport",
            "ruteplanleggjar",
            "ruteplanlegger",
            "route planner",
            "parkering",
            "parkering og hvileplasser",
            "parking and rest area",
            "reisegaranti",
            "reisegaranti",
            "travel guarantee",
            "tilrettelagd transport",
            "tilrettelagt transport",
            "transport services for disabled persons",
            "takstar og kjøpsinformasjon",
            "takster og kjøpsinformasjon",
            "fare and purchase information",
            "ruteinformasjon",
            "ruteinformasjon",
            "operational information",
            "miljøvenleg transport",
            "miljøvennlig transport",
            "environmental transportation",
            "næring",
            "næring",
            "business",
            "tilskotsordningar for næring",
            "tilskuddsordninger for næring",
            "public funding schemes for business",
            "næringsliv",
            "næringsliv",
            "agriculture",
            "næringsutvikling",
            "næringsutvikling",
            "business development",
            "landbruk",
            "landbruk",
            "agriculture sector",
            "handel og service",
            "handel og service",
            "trade and service",
            "avløysartilskot",
            "avløsertilskudd",
            "farm assistance grant",
            "skogbrukstilskot",
            "skogbrukstilskudd",
            "forestry grants",
            "tilskot til økologisk landbruk",
            "tilskudd til økologisk landbruk",
            "organic agriculture grants",
            "produksjonstilskot",
            "produksjonstilskudd",
            "produce grants",
            "stønad til bedrifter",
            "støtte til bedrifter",
            "support for businesses",
            "jordbrukstilskot",
            "jordbrukstilskudd",
            "farming grants",
            "tenester og krav til enkeltbransjar",
            "tjenester og krav til enkeltbransjer",
            "services and obligations for specific industries",
            "starte eller avslutte bedrift",
            "starte eller avslutte bedrift",
            "starting or closing a business",
            "rapportering for næringsdrivande",
            "rapportering for næringsdrivende",
            "business reporting",
            "standardar",
            "standarder",
            "standards",
            "tenester og krav til arbeidsgivar",
            "tjenester og krav til arbeidsgiver",
            "services and obligations for employers",
            "sertifisering og autorisasjon",
            "sertifisering og autorisasjon",
            "certification and authorization",
            "patent, design og varemerke",
            "patent, design og varemerke",
            "patent, design and trademarks",
            "utenlandsk bedrift",
            "utenlandsk bedrift",
            "foreign businesses in norway",
            "tilsyn og kontroll med næringsverksemd",
            "tilsyn og kontroll med næringsvirksomhet",
            "controlling compliance in businesses",
            "bevilling og løyve",
            "bevilling og tillatelser",
            "licenses and permissions",
            "stønad til bedrifter",
            "støtte til bedrifter",
            "support for businesses",
            "næringsklynger",
            "næringsklynger",
            "business parks",
            "næringsareal",
            "næringsareal",
            "business space",
            "dyrehelse",
            "dyrehelse",
            "animal health",
            "reindrift",
            "reindrift",
            "reindeer herding",
            "erstatningsordningar i landbruket",
            "erstatningsordninger i landbruket",
            "compensation schemes in agriculture",
            "vegbygging",
            "vegbygging",
            "road building",
            "skogbruk",
            "skogbruk",
            "forestry",
            "plantedyrking",
            "plantedyrking",
            "cultivation of plants",
            "plantevern",
            "plantevern",
            "plant protection",
            "landbrukseigedom",
            "landbrukseiendom",
            "agricultural property",
            "veterinær",
            "veterinær",
            "veterinarian",
            "produktsikkerheit",
            "produktsikkerhet",
            "product safety",
            "offentleg innkjøp",
            "offentlig innkjøp",
            "public procurement",
          ],
          "harvest": Object {
            "changed": Array [
              "2020-03-23T08:46:02+0100",
              "2020-03-23T13:01:06+0100",
              "2020-03-23T13:01:16+0100",
              "2020-03-23T13:01:55+0100",
              "2020-03-23T13:02:46+0100",
              "2020-03-23T13:03:48+0100",
              "2020-03-23T13:03:54+0100",
              "2020-03-23T13:04:00+0100",
              "2020-03-23T13:09:40+0100",
              "2020-03-23T13:09:46+0100",
              "2020-03-23T13:09:53+0100",
              "2020-03-23T13:10:00+0100",
              "2020-03-23T13:10:18+0100",
              "2020-03-23T13:10:36+0100",
              "2020-03-23T13:10:49+0100",
              "2020-03-23T13:10:56+0100",
              "2020-03-23T13:29:20+0100",
              "2020-03-23T13:30:16+0100",
              "2020-03-24T01:02:57+0100",
              "2020-03-25T01:02:53+0100",
              "2020-03-26T01:02:56+0100",
              "2020-03-27T01:02:54+0100",
              "2020-03-27T07:40:19+0100",
              "2020-03-27T07:40:26+0100",
              "2020-03-27T07:40:33+0100",
              "2020-03-27T07:40:42+0100",
              "2020-03-27T07:40:49+0100",
              "2020-03-27T07:40:56+0100",
              "2020-03-27T07:41:03+0100",
              "2020-03-27T07:41:10+0100",
              "2020-03-27T07:41:17+0100",
              "2020-03-27T07:41:25+0100",
              "2020-03-27T07:41:32+0100",
              "2020-03-27T07:41:40+0100",
              "2020-03-27T07:41:48+0100",
              "2020-03-27T07:41:57+0100",
              "2020-03-27T07:42:04+0100",
              "2020-03-27T07:42:12+0100",
              "2020-03-27T07:42:19+0100",
              "2020-03-27T07:42:26+0100",
              "2020-03-27T07:42:33+0100",
              "2020-03-27T07:42:41+0100",
              "2020-03-27T07:42:49+0100",
              "2020-03-27T07:42:57+0100",
              "2020-03-27T07:43:05+0100",
              "2020-03-27T07:43:12+0100",
              "2020-03-27T07:43:19+0100",
              "2020-03-28T01:03:14+0100",
              "2020-03-29T01:03:08+0100",
            ],
            "firstHarvested": "2020-03-23T08:46:02+0100",
            "lastChanged": "2020-03-29T01:03:08+0100",
            "lastHarvested": "2020-03-29T01:03:08+0100",
          },
          "hasCurrentnessAnnotation": Object {
            "hasBody": Object {
              "nb": "asd",
            },
            "inDimension": "http://iso.org/25012/2008/dataquality/Currentness",
          },
          "id": "522e4c94-2d38-428b-a24e-97d54f1b0213",
          "keyword": Array [
            Object {
              "nb": "test",
            },
            Object {
              "nb": "Åpne data",
            },
          ],
          "landingPage": Array [
            "https://www.aftenposten.no",
          ],
          "language": Array [
            Object {
              "code": "ENG",
              "prefLabel": Object {
                "en": "English",
                "nb": "Engelsk",
                "nn": "Engelsk",
                "no": "Engelsk",
              },
              "uri": "http://publications.europa.eu/resource/authority/language/ENG",
            },
          ],
          "losTheme": Array [
            Object {
              "children": Array [
                "https://psi.norge.no/los/tema/brann",
                "https://psi.norge.no/los/tema/forsvaret",
                "https://psi.norge.no/los/tema/varslingstjenester",
              ],
              "isTema": true,
              "losPaths": Array [
                "samfunnssikkerhet-og-beredskap",
              ],
              "name": Object {
                "en": "Civil protection and emergency planning",
                "nb": "Samfunnssikkerhet og beredskap",
                "nn": "Samfunnstryggleik og beredskap",
              },
              "synonyms": Array [],
              "uri": "https://psi.norge.no/los/tema/samfunnssikkerhet-og-beredskap",
            },
            Object {
              "children": Array [
                "https://psi.norge.no/los/ord/drivstoff-og-ladestasjoner",
                "https://psi.norge.no/los/ord/reisebillett",
                "https://psi.norge.no/los/ord/lokasjonstjenester",
                "https://psi.norge.no/los/ord/skoleskyss",
                "https://psi.norge.no/los/ord/ruteplanlegger",
                "https://psi.norge.no/los/ord/parkering-og-hvileplasser",
                "https://psi.norge.no/los/ord/reisegaranti",
                "https://psi.norge.no/los/ord/tilrettelagt-transport",
                "https://psi.norge.no/los/ord/takster-og-kjopsinformasjon",
                "https://psi.norge.no/los/ord/ruteinformasjon",
                "https://psi.norge.no/los/ord/miljovennlig-transport",
              ],
              "isTema": true,
              "losPaths": Array [
                "trafikk-og-transport/mobilitetstilbud",
              ],
              "name": Object {
                "en": "Mobility services",
                "nb": "Mobilitetstilbud",
                "nn": "Mobilitetstilbod",
              },
              "parents": Array [
                "https://psi.norge.no/los/tema/trafikk-og-transport",
              ],
              "synonyms": Array [
                "Kollektivtilbud",
                "Kollektivtilbod",
              ],
              "uri": "https://psi.norge.no/los/tema/mobilitetstilbud",
            },
            Object {
              "children": Array [
                "https://psi.norge.no/los/tema/tilskuddsordninger-for-naring",
                "https://psi.norge.no/los/tema/naringsliv",
                "https://psi.norge.no/los/tema/naringsutvikling",
                "https://psi.norge.no/los/tema/landbruk",
                "https://psi.norge.no/los/tema/handel-og-service",
              ],
              "isTema": true,
              "losPaths": Array [
                "naring",
              ],
              "name": Object {
                "en": "Business",
                "nb": "Næring",
                "nn": "Næring",
              },
              "synonyms": Array [],
              "uri": "https://psi.norge.no/los/tema/naring",
            },
          ],
          "modified": "2020-03-05T00:00:00+0100",
          "objective": Object {
            "nb": "Test åpne data",
          },
          "provenance": Object {
            "code": "VEDTAK",
            "prefLabel": Object {
              "en": "Governmental decisions",
              "nb": "Vedtak",
              "nn": "Vedtak",
            },
            "uri": "http://data.brreg.no/datakatalog/provinens/vedtak",
          },
          "publisher": Object {
            "id": "910244132",
            "name": "RAMSUND OG ROGNAN REVISJON",
            "orgPath": "/ANNET/910244132",
            "type": "no.dcat.datastore.domain.dcat.Publisher",
            "uri": "http://data.brreg.no/enhetsregisteret/enhet/910244132",
            "valid": false,
          },
          "references": Array [
            Object {
              "referenceType": Object {
                "code": "isReferencedBy",
                "prefLabel": Object {
                  "en": "Is Referenced By",
                  "nb": "Er referert av",
                  "nn": "Er referert av",
                },
                "uri": "http://purl.org/dc/terms/isReferencedBy",
              },
              "source": Object {
                "uri": "http://brreg.no/catalogs/910244132/datasets/1c941ffb-1849-44a6-ba06-ce6b1ddacc60",
              },
            },
          ],
          "source": "B",
          "subject": Array [
            Object {
              "definition": Object {
                "nb": "tema  i forbindelse med fastsetting av formues- og inntektsskatt som omhandler felles- formue, -gjeld, -inntekter og -kostnader som skattepliktige har knyttet til en eller flere boenheter i boligselskap eller boligsameie",
              },
              "identifier": "http://begrepskatalogen/begrep/2b29205f-e7e6-11e8-b7e4-005056821322",
              "prefLabel": Object {
                "nb": "boenheter i boligselskap eller boligsameie",
              },
              "uri": "https://fellesdatakatalog.brreg.no/api/concepts/f5507108-ee5f-4fa6-8e69-5c8c3ee540ec",
            },
          ],
          "theme": Array [
            Object {
              "id": "https://psi.norge.no/los/tema/samfunnssikkerhet-og-beredskap",
            },
            Object {
              "id": "https://psi.norge.no/los/tema/mobilitetstilbud",
            },
            Object {
              "id": "https://psi.norge.no/los/tema/naring",
            },
          ],
          "title": Object {
            "nb": "Test åpne data",
            "nn": "Nynorsk test",
          },
          "type": "dataset",
          "uri": "http://brreg.no/catalogs/910244132/datasets/e238edd9-b05c-435c-a844-89bb27d883bb",
        },
        Object {
          "harvest": Object {
            "firstHarvested": "2020-03-22T23:26:17",
            "lastChanged": "2020-03-22T23:26:17",
            "lastHarvested": "2020-03-29T12:10:27",
          },
          "harvestSourceUri": "http://fdk-nginx-search:8080/api/registration/apis/2260e2ed-d17f-4ff7-badf-9bc1eca0cc04",
          "id": "94b78a43-a2f9-41ea-a4c6-e13376b650a9",
          "publisher": Object {
            "name": "RAMSUND OG ROGNAN REVISJON",
            "orgPath": "PRIVAT/910244132",
            "prefLabel": Object {},
            "uri": "https://data.brreg.no/enhetsregisteret/api/enheter/910244132",
          },
          "schema": "{\\"$schema\\":\\"http://json-schema.org/draft-06/schema#\\",\\"$id\\":\\"https://fellesdatakatalog.brreg.no/informationmodels/2260e2ed-d17f-4ff7-badf-9bc1eca0cc04/schema\\",\\"definitions\\":{\\"Enhet\\":{\\"properties\\":{\\"organisasjonsnummer\\":{\\"type\\":\\"string\\"},\\"navn\\":{\\"type\\":\\"string\\"},\\"organisasjonsform\\":{\\"$ref\\":\\"#/definitions/Organisasjonsform\\"},\\"registreringsdatoEnhetsregisteret\\":{\\"type\\":\\"string\\"},\\"registrertIMvaregisteret\\":{\\"type\\":\\"boolean\\"},\\"naeringskode1\\":{\\"$ref\\":\\"#/definitions/Naeringskode\\"},\\"naeringskode2\\":{\\"$ref\\":\\"#/definitions/Naeringskode\\"},\\"naeringskode3\\":{\\"$ref\\":\\"#/definitions/Naeringskode\\"},\\"antallAnsatte\\":{\\"type\\":\\"integer\\"},\\"stiftelsedato\\":{\\"type\\":\\"string\\"},\\"institusjonellSektorkode\\":{\\"$ref\\":\\"#/definitions/Sektorkode\\"},\\"registrertIForetaksregisteret\\":{\\"type\\":\\"boolean\\"},\\"registrertIStiftelsesregisteret\\":{\\"type\\":\\"boolean\\"},\\"registrertIFrivillighetsregisteret\\":{\\"type\\":\\"boolean\\"},\\"sisteInnsendteAarsregnskap\\":{\\"type\\":\\"string\\"},\\"konkurs\\":{\\"type\\":\\"boolean\\"},\\"underAvvikling\\":{\\"type\\":\\"boolean\\"},\\"underTvangsavviklingEllerTvangsopplosning\\":{\\"type\\":\\"boolean\\"},\\"maalform\\":{\\"type\\":\\"string\\"},\\"_links\\":{\\"$ref\\":\\"#/definitions/Self\\"}}},\\"Naeringskode\\":{\\"properties\\":{\\"kode\\":{\\"type\\":\\"string\\"},\\"beskrivelse\\":{\\"type\\":\\"string\\"}}},\\"Sektorkode\\":{\\"properties\\":{\\"kode\\":{\\"type\\":\\"string\\"},\\"beskrivelse\\":{\\"type\\":\\"string\\"}}},\\"_Organisasjonsformer\\":{\\"properties\\":{\\"embedded\\":{\\"$ref\\":\\"#/definitions/Organisasjonsformer\\"}}},\\"Organisasjonsformer\\":{\\"properties\\":{\\"organisasjonsformer\\":{\\"type\\":\\"array\\",\\"items\\":{\\"$ref\\":\\"#/definitions/Organisasjonsform\\"}}}},\\"Organisasjonsform\\":{\\"properties\\":{\\"kode\\":{\\"type\\":\\"string\\"},\\"beskrivelse\\":{\\"type\\":\\"string\\"},\\"_links\\":{\\"$ref\\":\\"#/definitions/Self\\"}}},\\"Self\\":{\\"properties\\":{\\"self\\":{\\"$ref\\":\\"#/definitions/Href\\"}}},\\"Href\\":{\\"properties\\":{\\"href\\":{\\"type\\":\\"string\\"}}}}}",
          "title": Object {
            "nb": "Åpne Data fra Enhetsregisteret - API Dokumentasjon",
          },
          "type": "informationmodel",
        },
      ]
    }
  />
</Component>
`;
